.github/workflows/ci.yml

```diff
@@ -1,104 +1,225 @@
         uses: foundry-rs/foundry-toolchain@v1
         with:
           version: nightly
 
       - name: Run Foundry tests
-        run: set -o pipefail && yarn run:ci | tee ethereum.gasreport.ansi
+        run: yarn test:fuzz
+        env:
+          FOUNDRY_FUZZ_RUNS: "5000"
+
+  coverage:
+    needs: ["build", "lint"]
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+
+      - name: "Install Foundry"
+        uses: "foundry-rs/foundry-toolchain@v1"
+
+      - name: "Install lcov"
+        run: "sudo apt-get install lcov"
+
+      - name: "Generate the coverage report using the unit and the integration tests"
+        run: "yarn ci:coverage"
+
+      - name: "Upload coverage report to Codecov"
+        uses: "codecov/codecov-action@v3"
+        with:
+          files: "./lcov.info"
+          token: ${{ secrets.CODECOV_TOKEN }}
+
+      - name: "Add coverage summary"
+        run: |
+          echo "## Coverage result" >> $GITHUB_STEP_SUMMARY
+          echo "✅ Uploaded to Codecov" >> $GITHUB_STEP_SUMMARY
+
+  slither-analyze:
+    needs: ["build", "lint"]
+    runs-on: "ubuntu-latest"
+    permissions:
+      actions: "read"
+      contents: "read"
+      security-events: "write"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Compile foundry
+        run: forge build --build-info --skip */test/** */scripts/** --force
+
+      - name: "Run Slither analysis"
+        uses: "crytic/slither-action@v0.3.0"
+        id: "slither"
+        with:
+          ignore-compile: true
+          fail-on: "none"
+          sarif: "results.sarif"
+
+      - name: "Upload SARIF file to GitHub code scanning"
+        uses: "github/codeql-action/upload-sarif@v2"
+        with:
+          sarif_file: ${{ steps.slither.outputs.sarif }}
+
+      - name: "Add Slither summary"
+        run: |
+          echo "## Slither result" >> $GITHUB_STEP_SUMMARY
+          echo "✅ Uploaded to GitHub code scanning" >> $GITHUB_STEP_SUMMARY
+
+  check_storage_layout:
+    needs: ["build", "lint"]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
 
-      - name: Compare gas reports
-        uses: Rubilmax/foundry-gas-diff@v3.13.4
-        id: gas_diff
+      - name: Check storage layout
+        uses: Rubilmax/foundry-storage-check@v3.7
         with:
-          match: contracts/transmuter/DiamondProxy.sol
-          report: ethereum.gasreport.ansi
+          contract: contracts/transmuter/Layout.sol:Layout
+          rpcUrl: ${{ secrets.ETH_NODE_URI_MAINNET }}
+          address: "0x00253582b2a3FE112feEC532221d9708c64cEFAb"
```

Comment:
so you are using the on chain address, why would it fail the 1st time then?

Reply:
The address is an optional check as it will first compare it to the previous report

Reply:
Ok so it both check the previous and the on chain? Can you walk me through in vocal? Thanks